qiskit/extensions/quantum_initializer/isometry.py

```diff
@@ -73,6 +72,8 @@ def __init__(self, isometry, num_ancillas_zero, num_ancillas_dirty, epsilon=_EPS
         if len(isometry.shape) == 1:
             isometry = isometry.reshape(isometry.shape[0], 1)
 
+        self.iso_data = np.array(isometry)
```

Comment:
I don't believe it's necessary for this object to carry two copies of `isometry`; here and in `self.params[0]`. Also, the object should already be an ndarray.